contracts/src/v0.8/dev/BatchBHS.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+
+/**
+ * @title BatchBHS
+ * @notice The BatchBHS contract acts as a proxy to write many blockhashes to the
+ *   provided BlockhashStore contract efficiently in a single transaction. This results
+ *   in plenty of gas savings and higher throughput of blockhash storage, which is desirable
+ *   in times of high network congestion.
+ */
+contract BatchBHS {
+  BlockhashStore public immutable BHS;
+
+  constructor(address blockhashStoreAddr) {
+    BHS = BlockhashStore(blockhashStoreAddr);
+  }
+
+  /**
+   * @notice stores blockhashes of the given block numbers in the configured blockhash store, assuming
+   *   they are availble though the blockhash() instruction.
+   * @param blockNumbers the block numbers to store the blockhashes of. Must be available via the
+   *   blockhash() instruction, otherwise this function call will revert.
```

Comment:
Should the whole thing revert, or would it make sense to just skip over any that aren't available? Makes it a little safer in case only one is past the 256 block limit

Reply:
I think if we want to make it so that it doesn't revert, would we want to return the block numbers that failed to store? Maybe something like:

```solidity
// returns block numbers that we couldn't store
function store(uint256[] memory blockNumbers) public returns (uint256[]);
```

We could also silently skip over them, though in that case it might be harder to debug. E.g if one block number is too far back and we return that, the caller could call `storeVerifyHeader` with the farthest back block that we were able to store with this method.

Reply:
The advantage of having this revert right away is code and usage simplicity, I guess.

Reply:
Done.